File: fe-pc/ui/src/main/java/com/example/ui/service/UserService.java
package com.example.ui.service;

import com.example.ui.apis.UserApi;
import com.example.ui.config.ApiMethodsPrivate;
import org.json.JSONObject;

public class UserService {
    // Load thông tin
    public static JSONObject getInforUserService() {
        String url = UserApi.getInforUserApi;
        return ApiMethodsPrivate.getRequest(url);
    }
    
    // update thông tin
    public static JSONObject updateUserService(JSONObject body) {
        String url = UserApi.updateInforUserApi;
        return ApiMethodsPrivate.putRequest(url, body);
    }

}

File: fe-mobile/lifetipsui/app/src/main/java/com/example/lifetipsui/service/UserService.kt
package com.example.lifetipsui.service

import com.example.lifetipsui.apis.UserApi
import com.example.lifetipsui.helper.ApiMethodsPrivate
import kotlinx.serialization.json.JsonObject
import org.json.JSONObject

class UserService {
    companion object {
        // Hàm lấy thông tin người dùng
        suspend fun getInfor(): JSONObject? {
            return try {
                ApiMethodsPrivate.getRequest(UserApi.getInforApi)
            } catch (e: Exception) {
                println(">> [UserService] Error calling getInfor API: ${e.localizedMessage}")
                e.printStackTrace()
                null
            }
        }

        suspend fun updateInfor(data: Map<String, String>): JSONObject? {
            return try {
                val jsonBody = JSONObject()
                for ((key, value) in data) {
                    jsonBody.put(key, value)
                }

                val response = ApiMethodsPrivate.putRequest(UserApi.updateInforApi, jsonBody)
                response
            } catch (e: Exception) {
                println(">> [UserService] Error calling updateInfor API: ${e.localizedMessage}")
                e.printStackTrace()
                null
            }
        }

    }
}
